gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/longrunning/stub/HttpJsonOperationsStub.java

```diff
@@ -223,21 +230,58 @@ public static final HttpJsonOperationsStub create(
       TypeRegistry typeRegistry)
       throws IOException {
     return new HttpJsonOperationsStub(
-        OperationsStubSettings.newBuilder().build(), clientContext, callableFactory, typeRegistry);
+        OperationsStubSettings.newBuilder().build(),
+        clientContext,
+        callableFactory,
+        typeRegistry,
+        new HashMap<>());
+  }
+
+  // We choose to only override these two constructors
+  // The generator can generate these two combinations with custom HttpRules
+  public static final HttpJsonOperationsStub create(
+      ClientContext clientContext,
+      HttpJsonStubCallableFactory callableFactory,
+      Map<String, HttpRule> customHttpBindings)
+      throws IOException {
+    return new HttpJsonOperationsStub(
+        OperationsStubSettings.newBuilder().build(),
+        clientContext,
+        callableFactory,
+        TypeRegistry.getEmptyTypeRegistry(),
+        customHttpBindings);
+  }
+
+  public static final HttpJsonOperationsStub create(
+      ClientContext clientContext,
+      HttpJsonStubCallableFactory callableFactory,
+      TypeRegistry typeRegistry,
+      Map<String, HttpRule> customHttpBindings)
+      throws IOException {
+    return new HttpJsonOperationsStub(
+        OperationsStubSettings.newBuilder().build(),
+        clientContext,
+        callableFactory,
+        typeRegistry,
+        customHttpBindings);
   }
 
   /**
    * Constructs an instance of HttpJsonOperationsStub, using the given settings. This is protected
    * so that it is easy to make a subclass, but otherwise, the static factory methods should be
    * preferred.
    */
-  protected HttpJsonOperationsStub(OperationsStubSettings settings, ClientContext clientContext)
+  protected HttpJsonOperationsStub(
```

Comment:
Adding parameters to protected methods are breaking as well, as customers can extend this public class and override it.

Reply:
Ah. I'll try to overload the constructor then.